### DISCUSSION TITLE
Fluid images with next/image?

### DISCUSSION THREAD

**Discussion by JamieMason** [vercel/next.js#18327] [Help] [answered]
I'm trying to migrate a fluid hero image over to use `next/image` but am stuck for the moment. It's to replace the Hero over at https://trendygolf.com, which maintains its aspect ratio but is fluid. Its CLS is very low at 0.0001.

![Screen Recording 2020-10-27 at 19 07 31](https://user-images.githubusercontent.com/320492/97349910-f8473580-1887-11eb-829c-97cb3ed99880.gif)

```js
<Image src="/example.jpg" width={1200} height={566} alt="Example" priority />
```

The problem I'm having is that there's a fixed `width: 1200px` on a wrapping div which I can't seem to get around. Can anyone let me know where I'm going wrong here please? What I see instead is an image which cannot flex with the available space around it.

```html
<div style="max-width: 100%; width: 1200px">
  <div style="position: relative; padding-bottom: 47.16666666666667%">
    <img
      alt="Example"
      src="https://example.imgix.net/example.jpg?auto=format&amp;w=1200"
      srcset="
        https://example.imgix.net/example.jpg?auto=format&amp;w=320   320w,
        https://example.imgix.net/example.jpg?auto=format&amp;w=420   420w,
        https://example.imgix.net/example.jpg?auto=format&amp;w=768   768w,
        https://example.imgix.net/example.jpg?auto=format&amp;w=1024 1024w,
        https://example.imgix.net/example.jpg?auto=format&amp;w=1200 1200w
      "
      style="visibility: visible; height: 100%; left: 0; position: absolute; top: 0; width: 100%"
    />
  </div>
</div>
```

Thanks by the way for the amazing additions in Next v10 🎉

**Comment by jillmolloy** [comment]
@JamieMason this is pretty similar to the issue I'm having, here: https://github.com/vercel/next.js/discussions/18312

I'd suggest you wrap the Image element in a div and then you can target the div that the Image element creates via CSS using an important. 

For example, if you wrap it in a div with the class of `image-wrapper` you could do the following:

```css
.image-wrapper {
  > div {
    width: 100% !important;
  }
}
```

**Comment by zeekrey** [comment, marked as answer]
In Version **10.0.1** they added the `layout` prop to the image component. Setting it to `responsive` makes the above solution obsolete on my end.

**Reply by JamieMason** [reply]
Excellent, thanks @zeekrey. I was hoping there'd be an official solution to this 🎉

**Comment by luknl** [comment]
There are still some minor issues about this, at least for us! Please check [this link](https://github.com/vercel/next.js/discussions/18739#discussioncomment-121272) and let me know if some of you encounter same problem!
